### ISSUE TITLE
Enhancement: calculate register_datetime with ES

### ISSUE THREAD

**Issue by Dukestep** [ECCC-MSC/geomet-data-registry#45]
We are currently calculating the `register_datetime` property of a tile in the tileindex by using `elasticsearch.update_by_query` once the file is indexed. 

i.e File is registered in ES --> calculate datetime.now() --> update file's document in ES with `es.update_by_query()`.

I suggest that we look at a different approach that could use ElasticSearch's ingest pipelines to calculate `register_datetime` at the time the document is going to be indexed. This would avoid having to do call the `update_by_query` call and remove the need for the `refresh='wait_for'`  argument when we call `es.index()` for the received file.

We would have to set this pipeline during the initial creation of the index.

See: https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html and https://kb.objectrocket.com/elasticsearch/how-to-create-a-timestamp-field-for-an-elasticsearch-index-275

@tomkralidis, @RousseauLambertLP, thoughts on this?

**Comment by Dukestep** [comment]
I've implemented an approach that utilizes Elasticsearch's ingest pipelines in a local branch and there is definitely a HUGE net benefit in letting ES add the `register_datetime` property as it indexes a document.

I'm seeing thousandths of a second between indentifying a file and it getting registered in ES.

```json
    "properties": {
      "elevation": "2m",
      "identifier": "GDPS.ETA_TT-20190701000000-20190701000000",
      "reference_datetime": "2019-07-01T00:00:00",
      "file_creation_datetime": "2019-07-01T03:46:46.100000",
      "filepath": "/data-san/geomet/dev/feeds/model_gem_global/25km/grib2/lat_lon/00/000/CMC_glb_TMP_TGL_2_latlon.24x.24_2019070100_P000.grib2",
      "forecast_hour_datetime": "2019-07-01T00:00:00",
      "member": null,
      "identify_datetime": "2019-09-27T18:55:51.066768",
      "receive_datetime": "2019-09-27T18:55:50.893790",
      "layer": "GDPS.ETA_TT",
      "register_datetime": "2019-09-27T18:55:51.067903Z"
    }
```

Compare this to a result for a GDPS layer using the old method (almost 2 seconds!):
```json
          "properties": {
            "elevation": "100mb",
            "identifier": "GDPS.PRES_ES.100.3h-20190926120000-20190926120000",
            "reference_datetime": "2019-09-26T12:00:00",
            "file_creation_datetime": "2019-09-26T16:01:21.111000",
            "filepath": "/data/geomet/dev/feeds/model_gem_global/25km/grib2/lat_lon/12/000/CMC_glb_DEPR_ISBL_100_latlon.24x.24_2019092612_P000.grib2",
            "forecast_hour_datetime": "2019-09-26T12:00:00",
            "member": null,
            "identify_datetime": "2019-09-26T16:01:32.931020",
            "receive_datetime": "2019-09-26T16:01:32.927206",
            "layer": "GDPS.PRES_ES.100.3h",
            "register_datetime": "2019-09-26T16:01:34.859976"
          }
```

**Comment by tomkralidis** [comment]
Good analysis @Dukestep ! Issue a PR so we can analyze the changes?

**Comment by Dukestep** [comment]
Implemented in #50 and validated with AMQP feeds.
